### PR TITLE
viper: added environment supporting

### PR DESCRIPTION
**Pull request by arzonus**
I've had some problems with getting a password of Redis from the environment. The fix can help with it.

### CODE REVIEW

File: redigo.go [hunk 1]
```diff
@@ -44,51 +46,56 @@ func (b *Bundle) Build(builder *di.Builder) error {
...
+			// validating
+			if cfg.Host == "" {
+				return nil, errors.New("redis.host should be set")
 			}
 
-			// set default
-			if conf.MaxIdle == 0 {
-				conf.MaxIdle = 3
+			if cfg.MaxIdle < 0 {
+				return nil, errors.New("redis.MaxIdle should be greater then 0")
```
Comment (snovichkov) [inline redigo.go]: Invalid name of config key, should be redis.max_idle

File: redigo.go [hunk 2]
```diff
@@ -44,51 +46,56 @@ func (b *Bundle) Build(builder *di.Builder) error {
...
 			}
 
-			if conf.MaxActive < 0 {
+			if cfg.MaxActive < 0 {
 				return nil, errors.New("redis.MaxActive should be greater or equal to 0")
```
Comment (snovichkov) [inline redigo.go]: Invalid name of config key, should be redis.max_active

File: redigo.go [hunk 3]
```diff
@@ -44,51 +46,56 @@ func (b *Bundle) Build(builder *di.Builder) error {
...
 			}
 
-			if conf.IdleTimeout < 0 {
+			if cfg.IdleTimeout < 0 {
 				return nil, errors.New("redis.IdleTimeout should be greater or equal to 0")
```
Comment (snovichkov) [inline redigo.go]: Invalid name of config key, should be redis.idle_timeout

**Review by snovichkov — COMMENTED** [review]
Your pull request need some fixes, please fix them and I will merge this is pull request.

File: redigo.go (hunk 1 above)
Comment (arzonus) [reply redigo.go]: fixed

File: redigo.go (hunk 3 above)
Comment (arzonus) [reply redigo.go]: fixed

File: redigo.go (hunk 2 above)
Comment (arzonus) [reply redigo.go]: fixed

**Pull request merged**